### ISSUE TITLE
Double temp file (temp-temp-...) and double moving in SFTP (&FTP?) plugin

### ISSUE THREAD

**Issue by binwiederhier** [syncany/syncany#340]
The SFTP plugin currently uploads files to `<reporoot>/temp-temp-zGlzq-multichunk-75303cc708fe1479beb....` and then moves it to `<reporoot>/temporary/temp-zGlzq-multichunk-75303cc708fe1479beb...`. After that (when the TX is committed), the file is again moved to its final location at `<reporoot>/multichunks/multichunk-75303cc708fe1479beb...`.

The first indirection (`temp-temp-...`) is not necessary and should be removed to speed up the upload. Other plugins might have the same inefficiency.

**Comment by pimotte** [comment]
I'm not entirely sure if it is unnecessary. Though this might be counterintuitive: The first move ensures the atomicity of the upload(). The second move is for transaction-level atomicity. If we would not have the fist move, the following could happen:

1. Commit transaction.
2. Multichunk is uploaded. Fails halfway through. Action is not marked as started.
3. Upload is resumed. Attempt to upload file. File already exists. PANIC.

I don't know if the panic is real, though, if the ftp library handles this properly, there probably not a problem. I did write the Transaction stuff with the assumption that upload is atomic, for at least the above reason. (Maybe more, but I don't recall.) 

To sum up: The first move could probably be left out, but this is in my eyes not as straightforward as you phrase it. (Actually doing it is easier than being sure it results in correct code.)

**Comment by cr0** [comment]
Anything new on this? The dropbox plugin also uploads a temp-temp file.
